hack/local-e2e-common.sh

```diff
@@ -0,0 +1,474 @@
+#!/bin/bash
+# Common functions and setup for local e2e test scripts
+# This script is sourced by local-e2e-test.sh and local-e2e-pool-test.sh
+
+# Supported cloud platforms (keep in sync with Makefile test-e2e-local target)
+SUPPORTED_CLOUDS="aws/azure/gcp/ibmcloud/vsphere/nutanix/openstack"
+
+# Setup and validate KUBECONFIG
+# If KUBECONFIG environment variable is set, use it; otherwise use default ~/.kube/config
+function setup_kubeconfig() {
+    if [ -n "$KUBECONFIG" ]; then
+        echo "Using KUBECONFIG from environment: $KUBECONFIG"
+        if [ ! -f "$KUBECONFIG" ]; then
+            echo "Error: KUBECONFIG file not found: $KUBECONFIG" >&2
+            exit 1
+        fi
+    else
+        # Auto-detect: use default kubeconfig location if KUBECONFIG is not set
+        DEFAULT_KUBECONFIG="${HOME}/.kube/config"
+        if [ -f "$DEFAULT_KUBECONFIG" ]; then
+            export KUBECONFIG="$DEFAULT_KUBECONFIG"
+            echo "KUBECONFIG not set in environment, using default: $KUBECONFIG"
+        else
+            echo "Error: KUBECONFIG environment variable is not set and default kubeconfig not found at $DEFAULT_KUBECONFIG" >&2
+            echo "Please set KUBECONFIG environment variable:" >&2
+            echo "  export KUBECONFIG=/path/to/your/kubeconfig" >&2
+            exit 1
+        fi
+    fi
+}
+
+# Check cluster connectivity using oc or kubectl
+# Requires KUBECONFIG to be set (should be called after setup_kubeconfig)
+function check_cluster_connectivity() {
+    if [ -z "$KUBECONFIG" ]; then
+        echo "Error: KUBECONFIG is not set. Please call setup_kubeconfig() first." >&2
+        exit 1
+    fi
+    echo "Checking cluster connectivity with kubeconfig: $KUBECONFIG"
+    CLUSTER_ACCESSIBLE=false
```

Comment:
We don't need to use a var. Once connectivity is established, we can just `return` from this func.

Given that and my comment elsewhere about `oc` being mandatory, we should be able to simplify this function greatly.